File: xdm-core/src/main/java/xdm/core/util/StringUtils.java
package xdm.core.util;

import java.util.Locale;
import java.util.Objects;

public final class StringUtils {
  private StringUtils() {}

  public static boolean isNullOrEmpty(String str) {
    return str == null || str.isEmpty();
  }

  public static boolean isNullOrEmptyOrBlank(String str) {
    return str == null || str.trim().isEmpty();
  }

  public static byte[] getBytes(CharSequence sb) {
    return sb.toString().getBytes();
  }

  public static byte[] getBytes(String s) {
    return s.getBytes();
  }

  public static boolean containsIgnoreCase(String text, String match) {
    if (Objects.isNull(text) || Objects.isNull(match) || text.isEmpty() || match.isEmpty()) {
      return false;
    }
    return text.toLowerCase(Locale.ENGLISH).contains(match.toLowerCase(Locale.ENGLISH));
  }

  public static boolean equalsIgnoreCase(String str1, String str2) {
    if (str1 == null || str2 == null) {
      return false;
    }
    return str1.equalsIgnoreCase(str2);
  }
}

File: xdm-core/src/main/java/xdm/core/util/StringUtils.kt
package xdm.core.util

import java.util.*

object StringUtils {
    fun isNullOrEmpty(str: String?): Boolean {
        return str.isNullOrEmpty()
    }

    fun isNullOrEmptyOrBlank(str: String?): Boolean {
        return str.isNullOrBlank()
    }

    fun getBytes(sb: CharSequence): ByteArray {
        return sb.toString().toByteArray()
    }

    fun getBytes(s: String): ByteArray {
        return s.toByteArray()
    }

    fun containsIgnoreCase(text: String?, match: String?): Boolean {
        if (text.isNullOrBlank() || match.isNullOrBlank()) return false
        return text.contains(match, ignoreCase = true)
    }

    fun equalsIgnoreCase(str1: String?, str2: String?): Boolean {
        if (str1.isNullOrBlank() || str2.isNullOrBlank()) return false
        return str1.equals(str2, ignoreCase = true)
    }
}
